nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/groovy/org/apache/nifi/controller/queue/clustered/server/ConnectionLoadBalanceServerTest.groovy

```diff
@@ -0,0 +1,185 @@
+            logger.info("[${name?.toUpperCase()}] ${(args as List).join(" ")}")
+        }
+
+        tlsConfiguration = new TlsConfiguration(KEYSTORE_PATH, KEYSTORE_PASSWORD, KEYSTORE_TYPE, TRUSTSTORE_PATH, TRUSTSTORE_PASSWORD, TRUSTSTORE_TYPE)
+        sslContext = SslContextFactory.createSslContext(tlsConfiguration)
+    }
+
+    @Before
+    void setUp() {
+    }
+
+    @After
+    void tearDown() {
+        if (lbServer) {
+            lbServer.stop()
+        }
+    }
+
+    /**
+     * Asserts that the protocol versions in the parameters object are correct. In recent versions of Java, this enforces order as well, but in older versions, it just enforces presence.
+     *
+     * @param enabledProtocols the actual protocols, either in {@code String[]} or {@code Collection<String>} form
+     * @param expectedProtocols the specific protocol versions to be present (ordered as desired)
+     */
+    void assertProtocolVersions(def enabledProtocols, def expectedProtocols) {
+        if (CertificateUtils.getJavaVersion() > 8) {
+            assert enabledProtocols == expectedProtocols as String[]
+        } else {
+            assert enabledProtocols as Set == expectedProtocols as Set
+        }
+    }
+
+    @Test
+    void testRequestPeerListShouldUseTLS() {
+        // Arrange
+        logger.info("Creating SSL Context from TLS Configuration: ${tlsConfiguration}")
+        SSLContext sslContext = SslContextFactory.createSslContext(tlsConfiguration, SslContextFactory.ClientAuth.NONE)
+        logger.info("Created SSL Context: ${KeyStoreUtils.sslContextToString(sslContext)}")
+
+        def mockLBP = [
+                receiveFlowFiles: { Socket s, InputStream i, OutputStream o -> null }
+        ] as LoadBalanceProtocol
+        def mockER = [:] as EventReporter
+
+        lbServer = new ConnectionLoadBalanceServer(HOSTNAME, PORT, sslContext, NUM_THREADS, mockLBP, mockER, TIMEOUT_MS)
+
+        // Act
+        lbServer.start()
+
+        // Assert
+
+        // Assert that the default parameters (which can't be modified) still have legacy protocols and no client auth
+        def defaultSSLParameters = sslContext.defaultSSLParameters
+        logger.info("Default SSL Parameters: ${KeyStoreUtils.sslParametersToString(defaultSSLParameters)}" as String)
+        assertProtocolVersions(defaultSSLParameters.protocols, ["TLSv1.2", "TLSv1.1", "TLSv1"])
+        assert !defaultSSLParameters.needClientAuth
+
+        // Assert that the actual socket is set correctly due to the override in the LB server
+        SSLServerSocket socket = lbServer.serverSocket as SSLServerSocket
+        logger.info("Created SSL server socket: ${KeyStoreUtils.sslServerSocketToString(socket)}" as String)
+        assertProtocolVersions(socket.enabledProtocols, CertificateUtils.getCurrentSupportedTlsProtocolVersions())
+        assert socket.needClientAuth
+
+        // Clean up
+        lbServer.stop()
+    }
+
+    @Test
```

Comment:
May want to change to IT as GHA can encounter delays which might invalidate the assertions in this test.

Reply:
Added test case timing check.